gax-java/gax/src/main/java/com/google/api/gax/batching/BatcherImpl.java

```diff
@@ -374,6 +398,17 @@ public void close() throws InterruptedException {
       } else {
         throw new IllegalStateException("unexpected error closing the batcher", e.getCause());
       }
+    } catch (TimeoutException e) {
+      StringJoiner batchesStr = new StringJoiner(",");
+      for (Batch<ElementT, ElementResultT, RequestT, ResponseT> batch :
+          outstandingBatches.keySet()) {
+        batchesStr.add(batch.toString());
+      }
+      String msg = "Timed out trying to close batcher after " + timeout + ".";
+      msg += " Batch request prototype: " + prototype + ".";
+      msg += " Outstanding batches: " + batchesStr;
+
+      throw new BatchingException(msg);
```

Comment:
should I make this a more specific exception? ie. `BatcherCloseException`

Reply:
I think it depends on how do you plan to handle it in java-bigtable. If you want to rely on it to do some additional logics, then it makes sense to create a specific exception. If not, I don't think it is necessary.

Separately, depends on how to plan to handle this exception, you may want to logic the exception as well.

Reply:
I updated the intended usage in the description. I would prefer to avoid logging & throwing to avoid noise